WalletWasabi.Fluent/ViewModels/AddWallet/AddWalletPageViewModel.cs

```diff
@@ -93,9 +108,18 @@ private async Task OnImportWalletAsync()
 		}
 	}
 
-	private void OnRecoverWallet()
+	private async Task OnNextAsync()
 	{
-		Navigate().To(new WalletNamePageViewModel(WalletCreationOption.RecoverWallet));
+		var selected = SelectedOption ?? Options.First();
+
+		if (selected.CreationOption == WalletCreationOption.ImportWallet)
+		{
+			await ImportWalletAsync();
+		}
+		else
+		{
+			Navigate().To(new WalletNamePageViewModel(selected.CreationOption));
+		}
```

Comment:
Correct me, but I don't see how `SelectedOption` could be null.

```suggestion
		if (SelectedOption.CreationOption == WalletCreationOption.ImportWallet)
		{
			await ImportWalletAsync();
		}
		else
		{
			Navigate().To(new WalletNamePageViewModel(SelectedOption.CreationOption));
		}
```